### ISSUE TITLE
Trash does not show some items

### ISSUE THREAD

**Issue by dark-penguin** [mate-desktop/caja#564]
I have tmpfs mounted at /tmpfs via fstab. When I "Move to trash" files on that partition, a .Trash-<uid> directory appears with those files in it. However, I don't see those files when I open Trash, and those files are not deleted when I empty trash. I guess this may also affect other non-tmpfs mounted partitions under certain conditions.

While I'm at it, I want to plead the developers to provide any option to disable Trash partially or at least globaly. (Or does this have to be done in Gnome first as MATE is only a fork and does not implement new features?)

I have nothing against the general Trash idea, but there are circumstances when many people really want to disable it. I really hate the Trash folder appear on my removable drives and network-mounted shares, and I completely don't see the reason to have Trash on tmpfs (and with this bug, it just eats my RAM unless I remember to clean up manually). Creating a .Trash-<uid> file instead of a folder to prevent Trash from working is a very ugly fix for a lot of reasons (it's not system-wide, and it would require an on-mount script to create it every time tmpfs is mounted, just to name a few). I would rather have no Trash at all than have to deal with all this, but I don't even have an option to do that! Okay, maybe not an option, but at least some dirty fix via gsettings or something!.. And it would be ideal to have an option to disable Trash "everywhere except my home directory". I would even say this should be the default, seeing that vfs-mounted shares/drives do not have Trash (so why would "normally" mounted ones still have it?).
##

**Comment by miky2k** [comment]
Confirm bug still exist in mate 19.1

**Comment by lukefromdc** [comment]
The main trash too has a trash directory at  ~/.local/share/Trash and can only be used for files in the same filesystem (on the same device) as /home.  Remember that you cannot MOVE a file between filesystems. To move a file is to leave it physically in place and just change the name and/or path. This requires it be on the same filesystem. To "move to trash" from a different filesystem would require COPYING the file to trash.

Thus, a unified trash system across multiple directories has never to my knowledge existed in Linux, to create it would be quite a bit of work, it would cause trashing files on different directories than /home to increase space used for /home while not reducing space on the original directory unless move was folllowed by deleting the original. Lots of chances in all that for data loss, for bugs, and for tying up disk I/O remembering that some files (e.g. uncompressed video) can measure in the tens of GB.

I don't see this happening

**Comment by dark-penguin** [comment]
My suggestion was the opposite. Since properly supporting "Trash" anywhere other than the /home volume is too much trouble for too little benefit, just stop supporting it anywhere else! Currently, trash *is* supported on every filesystem, if it can create a .Trash folder in its root, and last time I checked, there was no way to empty that trash other than noticing that hidden folder and deleting it manually. It would be better to not have this .Trash at all, or at least have an option to disable it!

Oh, wait, I just upgraded to Buster, and now my tmpfs partition does not support Trash anymore... Nor does my cifs mount... Another ext4 partition does have Trash support, but items there are properly detected and emptied even after a system reboot! So this issue is actually fixed! I'm on MATE 1.20.4 , Debian Buster.
